test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterClaimControllerSpec.scala

```diff
@@ -65,344 +68,378 @@ class EnterClaimControllerSpec
+      compareUrlsWithouthId(
+        redirectLocation(result).getOrElse(fail()),
+        routes.EnterClaimController.enterClaim(UUID.randomUUID()).url
+      )
+    }
 
-        inSequence {
-          mockAuthWithNoRetrievals()
-          mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
-        }
+    "Redirect to the enterClaim page if we have an unfinished claim for a duty" in {
+      val taxCode              = TaxCode.A20
+      val dutiesSelectedAnswer = DutiesSelectedAnswer(Duty(taxCode))
+      val claim                = sample[Claim]
+        .copy(claimAmount = BigDecimal(10), paidAmount = BigDecimal(5), isFilled = false, taxCode = taxCode.value)
 
-        checkIsRedirect(
-          performAction(),
-          routes.EnterClaimController.checkClaim()
-        )
+      val session = getSessionWithPreviousAnswer(Some(ClaimsAnswer(claim)), Some(dutiesSelectedAnswer))._1
 
+      inSequence {
+        mockAuthWithNoRetrievals()
+        mockGetSession(session)
       }
 
-      "number of claims is more than duties selected" in {
-        def performAction(): Future[Result] = controller.startClaim()(FakeRequest())
-
-        val dutiesSelectedAnswer = DutiesSelectedAnswer(Duty(TaxCode.A00))
+      val result = performAction()
 
-        val claim1 = sample[Claim]
-          .copy(claimAmount = BigDecimal(10), paidAmount = BigDecimal(5), isFilled = false, taxCode = "A00")
-
-        val claim2 = sample[Claim]
-          .copy(claimAmount = BigDecimal(10), paidAmount = BigDecimal(5), isFilled = false, taxCode = "A00")
-
-        val answers = IncompleteClaimsAnswer(List(claim1, claim2))
-
-        val draftC285Claim = sessionWithClaimState(Some(answers))._3
+      status(result) shouldBe SEE_OTHER
+      compareUrlsWithouthId(
+        redirectLocation(result).getOrElse(fail()),
+        routes.EnterClaimController.enterClaim(UUID.randomUUID()).url
+      )
+    }
+  }
 
-        val updatedClaim = draftC285Claim.copy(dutiesSelectedAnswer = Some(dutiesSelectedAnswer))
+  "enterClaim page" must {
 
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
+    def performAction(id: UUID): Future[Result] = controller.enterClaim(id)(FakeRequest())
 
-        val updatedJourney = fillingOutClaim.copy(draftClaim = updatedClaim)
+    "redirect to the start of the journey if the session is empty" in {
+      val session = getSessionWithPreviousAnswer(None)._1
+      inSequence {
+        mockAuthWithNoRetrievals()
+        mockGetSession(session.copy(journeyStatus = None))
+      }
 
-        inSequence {
-          mockAuthWithNoRetrievals()
-          mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
-        }
+      checkIsRedirect(
+        performAction(UUID.randomUUID()),
+        baseRoutes.StartController.start()
+      )
+    }
 
-        checkIsRedirect(
-          performAction(),
-          routes.EnterClaimController.checkClaim()
-        )
+    //TODO finish error page
```

Comment:
TODO: ?

Reply:
There will be a new error page, Callum is working on it. Will come in the next 1-2 sprints